iocore/net/UnixNetVConnection.cc

```diff
@@ -1188,6 +1188,9 @@ UnixNetVConnection::populate(Connection &con_in, Continuation *c, void *arg)
     return EVENT_ERROR;
   }
 
+  // reset stale read triggered on keep alive connection to prevent getting into read ready list
+  this->read.triggered = 0;
```

Comment:
Refer to the code of `UnixNetVConnection::migrateToCurrentThread()`:

https://github.com/apache/trafficserver/blob/c196db5f0a5b957983351ad51ef360bfc68fbd66/iocore/net/UnixNetVConnection.cc#L1400-L1425

It detach SSL object from SSLNetVC and clear the `sslv->ssl` to `nullptr`.
A new `SSLNetVC` is created at L1416 and then `UnixNetVConnection::populate()` is called.
There is no SSL object leaks.

As a new `SSLNetVC`, the `this->read.triggered` must be `0`.

IMO, it is not the root cause of issue #7096 .

Reply:
Yeah, the problem is not on the new `SSLNetVC`. There's a race condition between thread A trying to migrate the VC from thread B, while thread B is doing all sorts of I/O, inactivity cop audit etc on the same VC in parallel. 

You are right that the read.triggered didn't really solve the crashes. The crashes returned after a few hours. I needed to guard with the VC's mutex on the new thread for the migrating VC to avoid the corruption. Will update the PR with the patch we've been running in a couple of days.